mars/services/task/execution/ray/executor.py

```diff
@@ -80,6 +80,22 @@ def _optimize_subtask_graph(subtask_graph):
     return _optimize_physical(subtask_graph)
 
 
+async def _cancel_ray_task(obj_ref, kill_timeout: int = 3):
+    ray.cancel(obj_ref, force=False)
+    try:
+        await asyncio.to_thread(ray.get, obj_ref, timeout=kill_timeout)
+    except ray.exceptions.TaskCancelledError:  # pragma: no cover
+        logger.info("Cancel ray task %s successfully.", obj_ref)
+    except BaseException as e:
+        logger.info(
+            "Failed to cancel ray task %s with exception %s, "
+            "force cancel the task by killing the worker.",
+            e,
+            obj_ref,
+        )
+        ray.cancel(obj_ref, force=True)
```

Comment:
> How long `ray.cancel` force take? Will this blocking event loop?

From the docs of ray we can know when force=True, the executing task will immediately exit. If the task is already finished, nothing will happen. For safety, it's better to wrap the `ray.cancel` with `asyncio.to_thread`. How about doing the change in the next pr because the ci takes too long?